vignettes (#74)
Genentech/jmpost: vignettes/Intro_to_jmpost.Rmd

---
title: "Introduction to `jmpost`"

> subtitle: "`J`oint `M`odel for `P`osterior `O`verall `S`urvival `T`ranjectories"

Reviewer: ```suggestion
subtitle: "`J`oint `M`odels for `P`redicting `O`verall `S`urvival `T`rajectories"
```

  - \usetikzlibrary{shapes.geometric, arrows}   
output:
  pdf_document: default
bibliography: bibliography.bib
vignette: |
  %\VignetteIndexEntry{Introduction to `jmpost`}
  %\VignetteEncoding{UTF-8}
  %\VignetteEngine{knitr::rmarkdown}
 
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  eval = F
)
```
# Introduction

In clinical development of new cancer medicines a key challenge is to decide on the next steps after having collected the initial data in Phase 1 studies. 
In phase 1 studies patients are followed until the cancer is growing again (progression event). 
Thus, reliable overall survival data are not available. 
However, we can use additional information from previous clinical trials or real-world data to understand the time between progression and death.
Specifically, we can correlate the tumor response data, that are longitudinal measurements, (i.e. the shrinkage and growth of tumor lesions over time) with the overall survival of the patients or their hazard ratios [@beyer2020multistate]. 
Thereby we can predict the overall survival in our Phase 1 studies and therefore make better decisions [@kerioui2020bayesian].

> The sum of the longest diameter of the cancer (SLD) provides an effective and representative biomarker for longitudinal measurements in solid cancer. 

Reviewer: ```suggestion
The sum of the longest diameters (SLD) of the cancer lesions provides an effective and representative biomarker for longitudinal measurements in solid cancer. 
```

> SLD is relatively easy to measure with great accuracy while also being a good representative of the development of cancer. 

Reviewer: ```suggestion
SLD is relatively easy to measure with reasonable accuracy while also being a good representative of the development of cancer. 
```

The association between the longitudinal data and the overall survival can be achieved in different ways.

> The predicted SLD value or its first derivative over time can be directly treated as the linking factor.

Reviewer: ```suggestion
The predicted SLD value or its first derivative over time can be directly treated as the linking factor in the hazard model for survival.
```

Bayesian methodology provides an important tool for the analysis of similar data.
Firstly, the use of historical data from previous trials is improving the performance of the models.
Secondly, prior definition allows the adjustment of the models according to the specific factors of each analysis.

Although these statistical methods provide significant advantages in decision making processes in cancer drug development, their implementation is relatively intricate and time consuming.
Compared to other R-packages such as `joineR` and `jmbayes`, `jmpost` provides the flexibility of non-linear mixed effects modeling for the longitudinal part of the joint model. 
Non-linear parametrization respects the biological implications of clinical trials in oncology where the observed tumor response usually follows non-linear patterns.

# Minimal usage 

The minimal usage of the `jmpost` package calls example stan code already provided in the repository of the package. 
The following chunk illustrates the absolute minimal usage of the package.

> ```{r, eval=FALSE}

Reviewer: maybe add 1-line comments above each command in this code example to further help the reader understand it?

my_long_mod <- LongMod()

my_temp_os <- LogLogisticOs()

my_link <- ttgLink()

my_os <- parametrize(osmod = my_temp_os, link = my_link)
JointModel(long = my_long_mod, os = my_os)
```

> The final two steps of the package workflow are not available yet.

Reviewer: ```suggestion
The final two steps of the package workflow are not available yet:
```

Thereby we can predict the overall survival in our Phase 1 studies and therefore make better decisions [@kerioui2020bayesian].

The sum of the longest diameter of the cancer (SLD) provides an effective and representative biomarker for longitudinal measurements in solid cancer. 
SLD is relatively easy to measure with great accuracy while also being a good representative of the development of cancer. 

The association between the longitudinal data and the overall survival can be achieved in different ways.
The predicted SLD value or its first derivative over time can be directly treated as the linking factor.
Other factors such as the time to growth, the tumor shrinkage rate or the tumor growth rate can also be used to associate longitudinal outcomes with overall survival.

Bayesian methodology provides an important tool for the analysis of similar data.
Firstly, the use of historical data from previous trials is improving the performance of the models.
Secondly, prior definition allows the adjustment of the models according to the specific factors of each analysis.

> Although these statistical methods provide significant advantages in decision making processes in cancer drug development, their implementation is relatively intricate and time consuming.

Reviewer: cite 2-3 other R-packages, like `jmbayes` or there is an INLA package for joint modeling

> Compared to other R-packages, `jmpost` provides the flexibility of non-linear modeling for the longitudinal part of the joint model. 

Reviewer: ```suggestion
Compared to other R-packages, `jmpost` provides the flexibility of non-linear mixed effects modeling for the longitudinal part of the joint model. 
```

Non-linear parametrization respects the biological implications of clinical trials in oncology where the observed tumor response usually follows non-linear patterns.

# Minimal usage 

> 

Reviewer: add some text

```{r}
my_long_mod <- LongModel(stan = StanModule(
    functions = "exp_long_functions.stan",
    data = "exp_long_data.stan",
    priors = long_prior(),
    inits = list(),
    parameters = "exp_long_parameters.stan",

>     transformed_parameters = "exp_long_transformed_parametes.stan"

Reviewer: can we avoid this for the minimal use?

))

my_temp_os <- LogLogisticOs()

my_link <- HazardLink(
    parameters = "beta_ttg",
    contribution = "* rep_matrix(ttg(psi_ks, psi_kg, psi_phi), rows(time))",

>     stan = StanModule()

Reviewer: same here?

)

my_os <- parametrize(osmod = my_temp_os, link = my_link)
JointModel(long = my_long_mod, os = my_os)
```

> 

Reviewer: add text here what is going to be added to the package, few sentences

> # Common usage

Reviewer: ```suggestion
# Customizing the analysis
```

           parameters = "",
           transformed_parameters = "",
           generated_quantities = "",
           priors = list(),
           inits = list()
)
```

The `jmpost` repository contains stan code for the creation of a specific stan model. 
The user is able to populate the longitudinal object `Long_mod` either by specifying the stan files containing the relevant code of the section or by directly typing the stan code into R. 
We split the model section of the stan file into two parts, the `priors` and the `model`. 
`Priors` contain a list with the names of the parameters and their prior distribution definitions while `model` is a character that contains model relevant information such us the definition of the likelihood.
The list of priors should be a named list of strings where each element represents the stan code for the definition of the desired density. 

```{r}
prior_set <- StanModule(priors = list("param_A" = "normal(0,1);"))
```

The user can modify the prior list at any point during the workflow. 

```{r}
prior_set@priors$param_A <- "normal(1,1);"
```

Call of the object prints the different sections.

```{r Long_model print}
Long_mod
```

> The overall survival object contails all ther information for the paramtetrisation of the second sub-model of the joint model. 

Reviewer: ```suggestion
The overall survival object contails all ther information for the parametrisation of the second sub-model of the joint model. 
```

Secondly, prior definition allows the adjustment of the models according to the specific factors of each analysis.

Although these statistical methods provide significant advantages in decision making processes in cancer drug development, their implementation is relatively intricate and time consuming.
Compared to other R-packages such as `joineR` and `jmbayes`, `jmpost` provides the flexibility of non-linear mixed effects modeling for the longitudinal part of the joint model. 
Non-linear parametrization respects the biological implications of clinical trials in oncology where the observed tumor response usually follows non-linear patterns.

# Minimal usage 

The minimal usage of the `jmpost` package calls example stan code already provided in the repository of the package. 

> The following chank illustrates the absolute minimal usage of the package.

Reviewer: ```suggestion
The following chunk illustrates the absolute minimal usage of the package.
```

The following chunk illustrates the absolute minimal usage of the package.

```{r, eval=FALSE}
# The longitudinal submodel
my_long_mod <- LongMod()
# The templated survival submodel
my_temp_os <- LogLogisticOs()

> # The link factor

Reviewer: ```suggestion
# The link between longitudinal and OS models
```
I would avoid "factor", I don't think it is a factor?
